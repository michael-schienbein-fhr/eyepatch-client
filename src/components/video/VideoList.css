.VideoList-container {
  /* display: flex; */
  min-height: calc(100vh - 234px);
  min-width: calc(30vw - 355px);
  justify-content: center;
  float: none;
  clear: both;
  width: 100%;
  position: relative;
  padding-bottom: 0.25em;
  padding-top: 0.25em;
  height: 0;
	flex-direction: column;
  overflow-y: auto;
  background-color: black;
  border: 1px solid;
  border-color: #6c757d;
  border-radius: .25rem;
  /* align-items: top; */
}
::-webkit-scrollbar {
		background: black;
		width: 8px;
}
::-webkit-scrollbar-button {
		display: none;
}
::-webkit-scrollbar-thumb {
		background: #141414;
		border-radius: 4px;
}
  .VideoList-container:hover::-webkit-scrollbar-thumb {
  background: #222222;
} 
  .VideoList-container:active::-webkit-scrollbar-thumb {
  background: #333333;
} 
