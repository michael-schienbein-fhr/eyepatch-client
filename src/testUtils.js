import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const demoQueueVideo = {
  username: 'testuser2',
  type: 'video',
  action: 'add',
  text: '"testVideoTitle" added to queue for user: "testuser2".',
  videoId: 'testVideoId',
  title: 'testVideoTitle',
  description: 'testDescription',
  thumbnail: 'testThumbnail'
}
const demoSearchVideo = { "kind": "youtube#searchResult", "etag": "3_5Vn-wWRzIN5QC7aOEWvZsmMwI", "id": { "kind": "youtube#video", "videoId": "GjbibR6DV1s" }, "snippet": { "publishedAt": "2015-08-11T17:03:57Z", "channelId": "UCi_ZH3bYHidyHsETH5hbTFQ", "title": "Dusky - Skin Deep (Official Video)", "description": "Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \"This EP was all ...", "thumbnails": { "default": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/default.jpg", "width": 120, "height": 90 }, "medium": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg", "width": 320, "height": 180 }, "high": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/hqdefault.jpg", "width": 480, "height": 360 } }, "channelTitle": "DuskyMusic", "liveBroadcastContent": "none", "publishTime": "2015-08-11T17:03:57Z" } }

const testRoom = {
  "id": 1,
  "roomOwner": "testadmin",
  "roomName": "testadmin",
  "roomMembers": null,
  "videoQueue": null,
  "hasPass": true,
  "created_at": "2021-10-22T10:34:59.443Z"
}
const UserProvider =
  ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider, demoUser, demoQueueVideo, demoSearchVideo, testRoom };

