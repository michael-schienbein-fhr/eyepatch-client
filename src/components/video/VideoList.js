import './VideoList.css';
import { useRef, useEffect } from 'react';
import VideoQueueItem from './VideoQueueItem';
import VideoSearchItem from './VideoSearchItem';
const VideoList = ({ connectionStatus, videoSearchRes, globalQueue, globalVideoId, handleVideoQueing, handleVideoRemoval, handleVideoChange }) => {
    const scrollTarget = useRef(null);

    useEffect(() => {
        if (scrollTarget.current) {
            scrollTarget.current.scrollIntoView({ behavior: 'smooth' });
        }
    }, [globalQueue.length]);
    const renderedVideos = (videoSearchRes.length > 0)
        ? videoSearchRes.map((video) => {
            return <VideoSearchItem
                key={video.id.videoId}
                video={video}
                handleVideoQueing={handleVideoQueing}
                connectionStatus={connectionStatus} />
        })
        : globalQueue.map((video) => {
            return <VideoQueueItem
                key={video.videoId}
                video={video}
                handleVideoRemoval={handleVideoRemoval}
                handleVideoChange={handleVideoChange}
                globalVideoId={globalVideoId}
                connectionStatus={connectionStatus} />
        });

    return (
        <div className='VideoList-container'>
            {renderedVideos}
            <div ref={scrollTarget} />
        </div>
    )
};
export default VideoList;