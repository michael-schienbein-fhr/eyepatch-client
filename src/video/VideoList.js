import './VideoList.css';
import { useRef, useEffect } from 'react';
import VideoItem from './VideoItem';
import VideoItem2 from './VideoItem2';
const VideoList = ({ connectionStatus, videoSearchRes, globalQueue, globalVideoId, handleVideoQueing, handleVideoRemoval, handleVideoChange }) => {
    const scrollTarget = useRef(null);

    useEffect(() => {
        if (scrollTarget.current) {
            scrollTarget.current.scrollIntoView({ behavior: 'smooth' });
        }
    }, [globalQueue.length]);
    const renderedVideos = (videoSearchRes.length > 0)
        ? videoSearchRes.map((video) => {
            return <VideoItem
                key={video.id.videoId}
                video={video}
                handleVideoQueing={handleVideoQueing}
                connectionStatus={connectionStatus} />
        })
        : globalQueue.map((video) => {
            return <VideoItem2
                key={video.videoId}
                video={video}
                handleVideoRemoval={handleVideoRemoval}
                handleVideoChange={handleVideoChange}
                globalVideoId={globalVideoId}
                connectionStatus={connectionStatus} />
        });

    return (
        <div className='VideoList-container'>
            {renderedVideos}
            <div ref={scrollTarget} />
        </div>
    )
};
export default VideoList;