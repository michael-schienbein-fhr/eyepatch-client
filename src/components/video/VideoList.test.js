import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { UserProvider, demoQueueVideo, demoSearchVideo } from "../../testUtils";
import VideoList from "./VideoList";

// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";
// const nock = require('nock')



it("matches snapshot", function () {
  window.HTMLElement.prototype.scrollIntoView = function () { };
  const { asFragment } = render(
    <UserProvider>
      <VideoList
        globalQueue={[demoQueueVideo]}
        videoSearchRes={[demoSearchVideo]}
      />
    </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
