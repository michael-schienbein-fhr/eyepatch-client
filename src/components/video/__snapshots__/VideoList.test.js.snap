// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="VideoList-container"
  >
    <div
      class=" video-item item"
      disabled=""
    >
      <img
        alt="Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \\"This EP was all ..."
        class="image"
        src="https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg"
      />
      <span>
        Dusky - Skin Deep (Official Video)
      </span>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
